cmd/crossplane/rbac/rbac.go

```diff
@@ -36,28 +35,19 @@ const (
 	ManagementPolicyBasic = string(rbac.ManagementPolicyBasic)
 )
 
-// Command configuration for the RBAC manager.
-type Command struct {
-	Name                string
-	Sync                time.Duration
-	LeaderElection      bool
-	ManagementPolicy    string
-	ProviderClusterRole string
-}
-
-// FromKingpin produces the RBAC manager command from a Kingpin command.
-func FromKingpin(cmd *kingpin.CmdClause) *Command {
-	c := &Command{Name: cmd.FullCommand()}
-	cmd.Flag("sync", "Controller manager sync period duration such as 300ms, 1.5h or 2h45m").Short('s').Default("1h").DurationVar(&c.Sync)
-	cmd.Flag("manage", "RBAC management policy.").Short('m').Default(ManagementPolicyAll).EnumVar(&c.ManagementPolicy, ManagementPolicyAll, ManagementPolicyBasic)
-	cmd.Flag("provider-clusterrole", "A ClusterRole enumerating the permissions provider packages may request.").StringVar(&c.ProviderClusterRole)
-	cmd.Flag("leader-election", "Use leader election for the conroller manager.").Short('l').Default("false").OverrideDefaultFromEnvar("LEADER_ELECTION").BoolVar(&c.LeaderElection)
+// TODO: Fix Enum Support and Remove Hardcoding
 
-	return c
+// Cmd starts Crossplane RBAC Manager controllers.
+type Cmd struct {
+	Sync                time.Duration `short:"s" default:"1h" help:"Controller manager sync period duration such as 300ms, 1.5h or 2h45m."`
+	LeaderElection      bool          `short:"l" default:"false" env:"LEADER_ELECTION" help:"Use leader election for the conroller manager."`
+	ManagementPolicy    string        `short:"m" name:"manage" default:"All" enum:"All,Basic" help:"RBAC management policy."`
+	ProviderClusterRole string        `help:"A ClusterRole enumerating the permissions provider packages may request."`
 }
 
 // Run the RBAC manager.
-func (c *Command) Run(log logging.Logger) error {
+func (c *Cmd) Run(zl *logr.Logger) error {
```

Comment:
You should be able to use `BindTo()` to receive an interface here rather than a pointer.

Reply:
@weastel Can you update here to receive an interface?

Reply:
@hasheddan @alecthomas I tried to receive an interface but got following error when running any command(For eg: `./crossplane core --debug`, `./crossplane core`):
```
crossplane: error: couldn't find binding of type logging.Logger for parameter 0 of core.Cmd.Run(), use kong.Bind(logging.Logger)
```

Is anything wrong in any implementations? If so can you please help me to resolve it.